pkg/operator/ceph/csi/secrets.go

```diff
@@ -212,9 +259,175 @@ func CreateCSISecrets(context *clusterd.Context, clusterInfo *client.ClusterInfo
 		return errors.Wrap(err, "failed to create kubernetes csi secret")
 	}
 
+	err = updateCephStatusWithCephxStatus(context, clusterInfo, clusterNamespaced, currentKeyCount)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// updateCephStatusWithCephxStatus updates the cephCluster status with CSI cephxStatus
+func updateCephStatusWithCephxStatus(context *clusterd.Context, clusterInfo *client.ClusterInfo, namespacedName types.NamespacedName, currentKeyCount int) error {
+	logger.Info("updating cephStatus with CSI cephxStatus")
+	err := retry.OnError(
```

Comment:
Other controllers use `OnConflict()` so this is probably good for consistency

```suggestion
	err := retry.OnConflict(
```